src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -135,31 +150,59 @@ static bool GetCommandAndKey(
                 // Example: "db.statement": SET;
                 activity.SetTag(SemanticConventions.AttributeDbStatement, command.Command);
             }
+        }
 
-            if (command.EndPoint != null)
+        if (command.EndPoint != null)
+        {
+            if (command.EndPoint is IPEndPoint ipEndPoint)
             {
-                if (command.EndPoint is IPEndPoint ipEndPoint)
-                {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerIp, ipEndPoint.Address.ToString());
-                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, ipEndPoint.Port);
-                }
-                else if (command.EndPoint is DnsEndPoint dnsEndPoint)
+                var ip = ipEndPoint.Address.ToString();
+                var port = ipEndPoint.Port;
+
+                activity?.SetTag(SemanticConventions.AttributeNetPeerIp, ip);
+                activity?.SetTag(SemanticConventions.AttributeNetPeerPort, port);
+
+                meterTags?.Add(SemanticConventions.AttributeServerAddress, ip);
+                meterTags?.Add(SemanticConventions.AttributeNetworkPeerAddress, ip);
+                if (port != DefaultPort)
                 {
-                    activity.SetTag(SemanticConventions.AttributeNetPeerName, dnsEndPoint.Host);
-                    activity.SetTag(SemanticConventions.AttributeNetPeerPort, dnsEndPoint.Port);
+                    meterTags?.Add(SemanticConventions.AttributeServerPort, port);
+                    meterTags?.Add(SemanticConventions.AttributeNetworkPeerPort, port);
                 }
-                else
+            }
+            else if (command.EndPoint is DnsEndPoint dnsEndPoint)
+            {
+                var host = dnsEndPoint.Host;
+                var port = dnsEndPoint.Port;
+
+                activity?.SetTag(SemanticConventions.AttributeNetPeerName, host);
+                activity?.SetTag(SemanticConventions.AttributeNetPeerPort, port);
```

Comment:
Candidate for separate PR - update semantic convention to 1.27.0. At least for network tags.
Maybe it will be worth to merge it before this PR? Then the final user experience and process review could be better.

Note. `net.peer.name` is deprecated. It it not only problem here.